### PR TITLE
Add UI portion of the batch ingest.

### PR DESCRIPTION
**Pull request by jordandukart**

### CODE REVIEW

**Comment by jordandukart** [comment]
Depends on https://github.com/Islandora/islandora_paged_content/pull/82, https://github.com/Islandora/islandora_solution_pack_book/pull/161 and https://github.com/Islandora/islandora_solution_pack_newspaper/pull/108

File: includes/form.inc [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    '#default_value' => islandora_get_namespace($object->id),
+    '#size' => 15,
+  );
+  unset($form['pdf_file']);
+  $form['target'] = array(
+    '#type' => 'managed_file',
+    '#title' => t('Zip file'),
+    '#description' => t('A Zip file containing a number of PDFs.'),
+    '#required' => TRUE,
+    '#upload_location' => 'public://',
```
Comment (willtp87) [inline includes/form.inc]: Can has file_default_scheme ?

File: includes/form.inc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+  }
+  if (count($models) > 1) {
+    $model = FALSE;
+  }
+  else {
+    $model = key($models);
+  }
+  $form = islandora_paged_content_upload_pdf_form($form, $form_state, $model);
+  // Remove some things we don't need from the base form.
```
Comment (willtp87) [inline includes/form.inc]: Not exactly what this is doing next.

File: includes/form.inc [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  );
+  $form['parent'] = array(
+    '#type' => 'value',
+    '#value' => $object->id,
+  );
+  $form['ingest_set_immediately'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Ingest immediately?'),
+    '#description' => t('If not selected, objects will just be preprocessed into the queue of objects to be ingested, to be fully-processed (and ingested) later--whenever an admin deems it an appropriate time.'),
+    '#default_value' => TRUE,
```
Comment (willtp87) [inline includes/form.inc]: This maybe should be off by default, as a few books could take a day.

File: includes/form.inc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  $predicate = $applicable_paged_cmodels[$form_state['values']['model']]['predicate'];
+  $parameters = array(
+    'target' => $form_state['values']['target'],
+    'text' => $form_state['values']['extract_text'],
+    'language' => $form_state['values']['language'],
+    'format' => $form_state['values']['image_format'],
+    'resolution' => $form_state['values']['resolution'],
+    'content_model' => $form_state['values']['model'],
+    'parent' => $form_state['values']['parent'],
+    'parent_relationship_uri' => 'info:fedora/fedora-system:def/relations-external#',
```
Comment (willtp87) [inline includes/form.inc]: Constant https://github.com/Islandora/tuque/blob/1.x/FedoraRelationships.php#L9

File: islandora_paged_content_pdf_batch.module [hunk 5]
```diff
@@ -13,6 +13,52 @@ function islandora_paged_content_pdf_batch_form_islandora_paged_content_upload_p
...
+}
+
+/**
+ * Access callback for the UI upload.
+ */
+function islandora_paged_content_pdf_batch_menu_access(AbstractObject $object) {
+  if (!islandora_object_access(ISLANDORA_INGEST, $object)) {
+    return FALSE;
+  }
+  $c = 'COLLECTION_POLICY';
```
Comment (willtp87) [inline islandora_paged_content_pdf_batch.module:45]: There is very similar code as what follows elsewhere in the pull and I wonder if we could have a helper in islandora_paged_content that returns the applicable parent or false if none. -- ignore this, code wasn't doing what I thought

**Pull request merged**